Mathlib/RingTheory/AlgebraicIndependent/TranscendenceBasis.lean

```diff
@@ -125,3 +208,299 @@ theorem IsTranscendenceBasis.isAlgebraic_field {F E : Type*} {x : ι → E}
   haveI : IsScalarTower (adjoin F S) (IntermediateField.adjoin F S) E :=
     IsScalarTower.of_algebraMap_eq (congrFun rfl)
   exact Algebra.IsAlgebraic.extendScalars (R := adjoin F S) (Subalgebra.inclusion_injective _)
+
+namespace AlgebraicIndependent
+
+variable (inj : Injective (algebraMap R A))
+
+section
+
+variable [NoZeroDivisors A]
+
+private def indepMatroid : IndepMatroid A where
```

Comment:
Do you need injectivity for this definition to work? It looks to me like if you drop the injectivity assumption, you will still have a matroid, but it will fail to be `Loopless`/`Simple` - see comments below. 

At least for linearly representable matroids, it is a mistake to insist that representations are injective  - for a start, it means they aren't closed under taking minors. I suspect something similar will be true here.

Reply:
Ok, I'm a little confused now. Is `{0}` an independent set in the matroid? If not, then it should be contained in the closure of the empty set, which would contradict your lemma about closures of subsingletons.

Reply:
If R->A is not injective, then A does not have any R-algebraically independent subset, so `(AlgebraicIndependent R ·)` doesn't define a matroid, see [AlgebraicIndependent.algebraMap_injective](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/AlgebraicIndependent/Basic.html#AlgebraicIndependent.algebraMap_injective). One can require R to be a subring of A, or consider the matroid defined by `(AlgebraicIndependent (range (algebraMap R A)) ·)` instead, but then the defeq becomes bad.

Reply:
Ah right, I misunderstood what the injectivity was doing. 

But what about my zero question?

Reply:
{0} is algebraic independent over R iff R is a singleton.